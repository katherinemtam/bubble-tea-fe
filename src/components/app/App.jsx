import React from 'react';
import { Switch, Route } from 'react-router-dom';
import DetailPage from '../details/DetailPage';
import ToppingsList from '../toppings/ToppingsList';

export default function App() {
  return (
    <>
      <Switch>
        <Route exact path="/" component={ToppingsList}/>
        <Route exact path="/topping/:id" component={DetailPage} />
      </Switch>
    </>
  );
}
