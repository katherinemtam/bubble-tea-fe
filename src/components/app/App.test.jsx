import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import App from './App';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<MemoryRouter><App /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
