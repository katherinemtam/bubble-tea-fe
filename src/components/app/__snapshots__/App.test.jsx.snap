// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <header
    class="Header"
  >
    <div>
      <img
        alt="logo"
        src="bubble-logo-home.png"
      />
      <h1>
        Bubble Tea Shop (Under Construction)
      </h1>
    </div>
    <nav>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
      <a
        href="/toppings"
      >
        All Toppings
      </a>
      <a
        href="/toppings/add"
      >
        Add Topping
      </a>
    </nav>
  </header>
  <main
    class="Home"
  >
    <h1>
      Welcome to the Bubble Tea Shop
    </h1>
    <img
      alt="logo"
      src="bubble-tea-home.jpg"
    />
    <p>
      While we're still under construction, checkout the toppings we'll be offering!
    </p>
    <a
      href="/toppings"
    >
      Check out all the toppings!
    </a>
  </main>
</DocumentFragment>
`;
